kong/plugins/datadog/log.lua

```diff
@@ -0,0 +1,71 @@
+local statsd_logger = require "kong.plugins.datadog.statsd_logger"
+
+local ngx_log = ngx.log
+local ngx_timer_at = ngx.timer.at
+local string_gsub = string.gsub
+
+local _M = {}
+
+local METRICS = { "request_count", "latency", "request_size", "status_count" }
+
+local function request_counter(api_name, logger)
+  local stat = api_name..".request.count"
+  logger:counter(stat, 1, 1)
+end
+
+local function status_counter(api_name, message, logger)
+  local stat = api_name..".request.status."..message.response.status
+  logger:counter(stat, 1, 1)
+end
+
+local function request_size_gauge(api_name, message, logger)
+  local stat = api_name..".request.size"
+  logger:gauge(stat, message.request.size, 1)
+end
+
+local function latency_gauge(api_name, message, logger)
+  local latency = message.response.response_received - message.started_at
+  local stat = api_name..".latency"
+  logger:gauge(stat, latency, 1)
+end
+
+local function log(premature, conf, message, logger)
+
+  local logger, err = statsd_logger:new(conf)
+  if err then
+    ngx_log(ngx.ERR, "failed to create Statsd logger: ", err)
+    return
+  end
+  
+  local metrics = conf.metrics
+  if not conf.metrics then
```

Comment:
It is the roles of the schemas to make sure a default value is available. Here your logic file also implements default configuration options, which is not a very good separation of concerns. You should specify a `default` in your schema, and implement a small `func` check that makes sure the given table has a length > 0.